### ISSUE TITLE
Bug when fdb craps itself

### ISSUE THREAD

**Issue by dubslow** [MersenneForum/MersenneForumAliquot#21]
Posting this mostly for the comedy factor. Testing the merge verifier and this randomly happened. Never, ever seen the db do this before.

```
./allseq.py 1251156 1266612 1406166 351930 1506036 688422
14:38:23 INFO root:114: 2018-02-26 14:38:23
14:38:23 INFO root:62: Initializing
14:38:23 INFO mfaliquot.application:442: Lock acquired, reading ../website/generated/AllSeq.json
14:38:23 INFO mfaliquot.application.updater:85: Checking ./allseq.drops.txt for seqs to drop
14:38:23 INFO mfaliquot.application.updater:116: Added 5 new seqs: 1251156 1266612 1406166 351930 1506036
14:38:24 INFO mfaliquot.application.updater:314: Updater init complete, starting FDB queries on 6 sequences
14:38:24 INFO mfaliquot.application.fdb:156: Seq 1251156: no basic information!
14:38:24 INFO mfaliquot.application.fdb:157: Seq 1251156: retrying query (4 tries left)
14:38:29 INFO mfaliquot.application.fdb:156: Seq 1251156: no basic information!
14:38:29 INFO mfaliquot.application.fdb:157: Seq 1251156: retrying query (3 tries left)
14:38:35 INFO mfaliquot.application.fdb:156: Seq 1251156: no basic information!
14:38:35 INFO mfaliquot.application.fdb:157: Seq 1251156: retrying query (2 tries left)
14:38:40 INFO mfaliquot.application.fdb:156: Seq 1251156: no basic information!
14:38:40 INFO mfaliquot.application.fdb:157: Seq 1251156: retrying query (1 tries left)
14:38:45 ERROR mfaliquot.application.fdb:153: Seq 1251156: bad data after 5 tries: Seq 1251156: no basic information!
14:38:45 ERROR mfaliquot.application.updater:224: Seq 1251156: no basic information!
14:38:45 INFO mfaliquot.application.updater:225: Skipping sequence 1251156
14:38:46 INFO mfaliquot.application:532: seqinfo written, lock released
14:38:46 ERROR root:109: allseq.py interrupted by TypeError: '<' not supported between instances of 'int' and 'NoneType'
Traceback (most recent call last):
  File "./allseq.py", line 107, in <module>
    main()
  File "./allseq.py", line 96, in main
    quitting = inner_main(updater, seqinfo, special)
  File "./allseq.py", line 66, in inner_main
    quitting = updater.do_all_updates(seqinfo, special)
  File "/home/bill/mfaliquot/mfaliquot/application/updater.py", line 317, in do_all_updates
    count, terminated = self.primary_update_loop(seqs_todo)
  File "/home/bill/mfaliquot/mfaliquot/application/updater.py", line 260, in primary_update_loop
    self.seqinfo.push_new_info(ali)
  File "/home/bill/mfaliquot/mfaliquot/application/__init__.py", line 588, in push_new_info
    self._heap.push(self._make_heap_entry(ali))
TypeError: '<' not supported between instances of 'int' and 'NoneType'
14:38:46 INFO mfaliquot:156: Sending logging email with 3 records
```

**Comment by dubslow** [comment]
This only happens when adding new sequences to the list. Normally, when info can't be gotten, the code just returns the old `ali` object, but adding a new sequence means no such old object exists of course.

@ChristianBeer What should the behavior be in this case? Add an otherwise-empty/garbage `ali` object, with high priority to be requeried, or just forget the sequence entirely and require the user to manually add the sequence again?

**Comment by ChristianBeer** [comment]
Adding sequences to the list should be a fire and forget operation. So the scripts need to deal with any errors. So in this case I would like to have an empty/dummy `ali` object that get's queried with the next try. When we extend the list again we just run a script that adds a couple of sequences every hour or so and don't want to check if the sequence was added for sure or an error happened.

**Comment by dubslow** [comment]
:+1:

**Comment by dubslow** [comment]
Turns out I misunderstood the bug. The code had been written long ago to keep the old `ali` object in case of network/fdb problems, and that code was no different for "new" sequences. Rather, the bug was in the details of how I managed "deleting" a sequence from the underlying min-heap.  I was using a `None` to be the "sequence" when the heap entry was to be deleted, but in the rare case of new-seq-with-no-data, that would lead to the None being directly compared to other valid sequences, which generates the error in OP. Fix is to replace None with Infinity.
